### ISSUE TITLE
Unable to build on Fedora 28

### ISSUE THREAD

**Issue by rkosegi** [HubSpot/NioImapClient#63]
Project can't be built on Fedora (and possibly on CentOS/RHEL too) due to compilation error:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] An unknown compilation problem occurred
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10.915 s
[INFO] Finished at: 2018-07-18T14:00:24+02:00
[INFO] Final Memory: 36M/510M
[INFO] ------------------------------------------------------------------------
```

when I run maven using `-X` it yields root cause (stripped only to relevant part):

```
-g -target 1.8 -source 1.8 -encoding UTF-8 -bootclasspath /etc/alternatives/jre/jre/lib/rt.jar:/etc/alternatives/jre/jre/lib/jce.jar:/etc/alternatives/jre/../classes/classes.jar
[DEBUG] incrementalBuildHelper#beforeRebuildExecution
[INFO] Compiling 123 source files to /home/rkosegi/git/ezz/NioImapClient/target/classes
Fatal Error: Unable to find package java.lang in classpath or bootclasspath
[DEBUG] incrementalBuildHelper#afterRebuildExecution
```

Important part is `/etc/alternatives/jre/jre/lib/rt.jar` (`jre` is twice in path, actual path should be just `/etc/alternatives/jre/lib/rt.jar`).

This comes from basepom parent (maven compiler plugin configuration):

```
<plugin>
  <artifactId>maven-compiler-plugin</artifactId>
  <configuration>
    <compilerArguments children.combine="append">
      <bootclasspath>${project.jdk7.home}/lib/rt.jar:${project.jdk7.home}/jre/lib/jce.jar:${project.jdk7.home}/../classes/classes.jar</bootclasspath>
    </compilerArguments>
  </configuration>
</plugin>
```

Question here why you configure it that way?

**Comment by jhaber** [comment]
Should be fixed by #64

**Comment by rkosegi** [comment]
@jhaber build passing now, thanks
